src/LimitedLDLFactorizations.jl

```diff
@@ -383,16 +362,48 @@ function lldl_factorize!(
   return S
 end
 
+"""
+    lldl(A)
+
+Compute the limited-memory LDLᵀ factorization of `A` without pivoting.
+`A` should be a lower triangular matrix.
+
+# Arguments
+- `A::SparseMatrixCSC{Tv,Ti}`: matrix to factorize (its strict lower triangle and
+                               diagonal will be extracted)
+
+# Keyword arguments
+- `P::AbstractVector{<:Integer} = amd(A)`: permutation vector.
+- `memory::Int=0`: extra amount of memory to allocate for the incomplete factor `L`.
+                   The total memory allocated is nnz(T) + n * `memory`, where
+                   `T` is the strict lower triangle of A and `n` is the size of `A`.
+- `α::Tv=Tv(0)`: initial value of the shift in case the incomplete LDLᵀ
+                 factorization of `A` is found to not exist. The shift will be
+                 gradually increased from this initial value until success.
+- `droptol::Tv=Tv(0)`: to further sparsify `L`, all elements with magnitude smaller
+                       than `droptol` are dropped.
+- `check_tril::Bool = true`: check if `A` is a lower triangular matrix. 
+"""
 function lldl(
-  T::SparseMatrixCSC{Tv, Ti},
-  adiag::AbstractVector{Tv},
-  P::AbstractVector{<:Integer};
+  A::SparseMatrixCSC{Tv, Ti};
+  P::AbstractVector{<:Integer} = amd(A),
   memory::Int = 0,
   α::Tv = Tv(0),
   droptol::Tv = Tv(0),
+  check_tril::Bool = true,
 ) where {Tv <: Number, Ti <: Integer}
-  S = LimitedLDLFactorization(T, adiag, P; memory = memory, α = α)
-  lldl_factorize!(S, T, adiag, droptol = droptol)
+  T = (!check_tril || istril(A)) ? A : tril(A)
+  S = LimitedLDLFactorization(T; P = P, memory = memory, α = α)
+  lldl_factorize!(S, T, droptol = droptol)
+end
+
+lldl(A::Array{Tv, 2}; kwargs...) where {Tv <: Number} = lldl(sparse(A); kwargs...)
+
+# symmetric matrix input
```

Comment:
Can you add a wrapper for `Hermitian` too?

Reply:
I'm not sure that complex matrices are supported for now:
https://github.com/JuliaSmoothOptimizers/LimitedLDLFactorizations.jl/blob/a824f6b2b0d5ef2daba127cece6e6a7a7c0f49a4/src/LimitedLDLFactorizations.jl#L29

Reply:
The matrix is Hermitian even if it's a real matrix :)

Reply:
I will check the other part of the PR asap